include/deal.II/base/smartpointer.h

```diff
@@ -340,16 +340,19 @@ template <typename T, typename P>
 inline SmartPointer<T, P> &
 SmartPointer<T, P>::operator=(T *tt)
 {
-  // optimize if no real action is
-  // requested
+  // optimize if no real action is requested
   if (t == tt)
     return *this;
 
+  // Let us unsubscribe from the current object
   if (pointed_to_object_is_alive && t != nullptr)
     t->unsubscribe(&pointed_to_object_is_alive, id);
+
+  // Then reset to the new object, and subscribe to it
   t = tt;
   if (tt != nullptr)
-    tt->subscribe(&pointed_to_object_is_alive, id);
+    t->subscribe(&pointed_to_object_is_alive, id);
```

Comment:
I found this particularly hard to read. We call `tt->` which seems to suggest that `tt` (the rhs of `operator=`) is subscribing to the object pointed to. Of course we want the *lhs* to do that, which is the `t` object. (I should also in a follow-up rename these variables!)

The thing is that `tt->` translates to `tt.t->`, and after `t=tt`, the objects pointed to are the same. In other words, the two lines are the same, but I think it is much easier to read when you say `t->`.